pylint/reporters/reports_handler_mix_in.py

```diff
@@ -12,17 +13,19 @@ class ReportsHandlerMixIn:
     related methods for the main lint class
     """
 
-    def __init__(self):
-        self._reports = collections.defaultdict(list)
-        self._reports_state = {}
+    def __init__(self) -> None:
+        self._reports: DefaultDict = collections.defaultdict(list)
+        self._reports_state: Dict = {}
```

Comment:
We should probably add typing of the Dict if we are taking the effort of adding typing. I could take a look if you want me to!

Reply:
Sure thing !

Reply:
```suggestion
        self._reports_state: Dict[str, bool] = {}
```